# Day 1 — Descriptive Statistics 

Computing and interpreting the center and spread of the Fare column from the Titanic dataset.

## Step 1 — Load and Inspect the Dataset

In [2]:
import pandas as pd

df = pd.read_csv ("tested.csv")

# Information on the dataset
print(df.shape)
df.info()

(418, 12)
<class 'pandas.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    str    
 4   Sex          418 non-null    str    
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    str    
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     str    
 11  Embarked     418 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 39.3 KB


### Missing Values
Check how many values are missing per column.

In [3]:
df.isnull().sum()


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age             86
SibSp            0
Parch            0
Ticket           0
Fare             1
Cabin          327
Embarked         0
dtype: int64

Dropped the one row with a missing Fare value, since only a single value is missing out of 418 rows — too small to affect the statistics either way.

In [4]:
df = df.dropna(subset=["Fare"])

## Step 2 — Compute Descriptive Statistics
Select the Fare column and compute its mean, median, mode, standard deviation, and IQR.

In [5]:
fare_col = df["Fare"]


In [6]:
mean_val = fare_col.mean()
print (f"Mean = ", mean_val)

median_val= fare_col.median()
print (f"Median = ",median_val )

mode_val = fare_col.mode()[0]
print(f"Mode = " , mode_val)

std_val =fare_col.std()
print(f"Standard Deviation = ",std_val) 

q1 = fare_col.quantile(0.25)
q3 = fare_col.quantile(0.75)

IQR = q3 - q1
print (f"IQR = ", IQR)


Mean =  35.627188489208635
Median =  14.4542
Mode =  7.75
Standard Deviation =  55.907576179973844
IQR =  23.6042


## Step 3 — Mean vs. Median: Which Represents a Typical Fare?

The mean (35.63) and median (14.45) are far apart — a difference of about 21. This large gap signals that the Fare column is heavily skewed by a small number of passengers who paid very high fares, which pulls the mean well above where most fares actually fall.

For this reason, the median is the more honest representation of a "typical" fare. The mean is distorted by outliers, while the median reflects the fare that a passenger in the middle of the distribution actually paid.